api/filters/service_instance_filter.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filters
+
+import (
+	"fmt"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+	"net/http"
+)
+
+const platformIDProperty = "platform_id"
+
+const ServiceInstanceValidationFilterName = "ServiceInstanceValidationFilter"
+
+// ServiceInstanceValidationFilter checks patch request for service offerings and plans include only label changes
+type ServiceInstanceValidationFilter struct {
+}
+
+func (*ServiceInstanceValidationFilter) Name() string {
+	return ServiceInstanceValidationFilterName
+}
+
+func (*ServiceInstanceValidationFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
+	platformID := gjson.GetBytes(req.Body, platformIDProperty).Str
+
+	if platformID != "" {
+		return nil, &util.HTTPError{
+			ErrorType:   "BadRequest",
+			Description: fmt.Sprintf("Providing %s property during provisioning/updating of a service instance is forbidden", platformIDProperty),
```

Comment:
What if I provide `service-manager`? Should this be allowed? We could just ignore whatever is provided and overwrite it with `service-manager`?

Reply:
addressed